### PR TITLE
Adding logic to Riemann clinet and trace to query processor interceptor

### PR DESCRIPTION
**Pull request by nbozic-zz**

### CODE REVIEW

File: src/main/java/io/smartcat/cassandra/diagnostics/report/RiemannQueryReporter.java [hunk 1]
```diff
@@ -56,17 +56,18 @@ public RiemannQueryReporter(Configuration config) {
...
-      riemannClient().event()
-        .service(serviceName)
-        .state("ok")
-        //.time(queryReport.startTime / 1000000)
-        .metric(queryReport.executionTime / 1000000f) // Log in ms so that we don't drown in high numbers
-        .attribute("client", queryReport.clientAddress)
-        .attribute("statement", queryReport.statement)
-        .send();
+    if (riemannClient() == null) {
+      logger.warn("Riemann client is not initialized!");
```
Comment (mgobec) [inline src/main/java/io/smartcat/cassandra/diagnostics/report/RiemannQueryReporter.java:60]: return after this. its pointless to send events with null client

**Comment by nivancevic** [comment]
The idea in `QueryProcessorWrapper::report` is to have as little activities as possible and to postpone everything to the executor service, that it, to the a new thread of execution (of lesser priority), in order to lower the performance impact to the original code. That was the reason why the construction of `QueryReport` is kept inside of `executor.submit()`.

File: src/main/java/io/smartcat/cassandra/diagnostics/QueryProcessorWrapper.java [hunk 2]
```diff
@@ -123,11 +123,14 @@ private ResultMessage originalProcessStatement(CQLStatement statement, QueryStat
...
   private void report(final long startTime, final long execTime, final CQLStatement statement,
       final QueryState queryState, final QueryOptions options, final String errorMessage) {
+      QueryReport report = new QueryReport(startTime,
+            execTime,
+            queryState.getClientState().getRemoteAddress().toString(),
+            statement.getClass().getSimpleName());
+
+    logger.trace("Reporting slow query: {}.", report);
     executor.submit(() -> {
-      reporter.report(new QueryReport(startTime,
```
Comment (nivancevic) [inline src/main/java/io/smartcat/cassandra/diagnostics/QueryProcessorWrapper.java]: The idea in QueryProcessorWrapper::report is to have as little activities as possible and to postpone everything to the executor service, that it, to the a new thread of execution (of lesser priority), in order to lower the performance impact to the original code. That was the reason why the construction of QueryReport is kept inside of executor.submit().

Comment (nivancevic) [inline src/main/java/io/smartcat/cassandra/diagnostics/QueryProcessorWrapper.java:127]: The idea in QueryProcessorWrapper::report is to have as little activities as possible and to postpone everything to the executor service, that it, to the a new thread of execution (of lesser priority), in order to lower the performance impact to the original code. That was the reason why the construction of QueryReport is kept inside of executor.submit().

Comment (nbozic-zz) [inline src/main/java/io/smartcat/cassandra/diagnostics/QueryProcessorWrapper.java:127]: I have added task to do that and will leave this just to have TRACE.

**Pull request merged**